app/main.py

```diff
@@ -1,13 +1,24 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    return {
+        "line": error["line_number"],
+        "column": error["column_number"],
+        "message": error["text"],
+        "name": error["code"],
+        "source": "flake8",
+    }
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {
+        "path": file_path,
+        "status": ("passed" if len(errors) == 0 else "failed"),
+        "errors": [(format_linter_error(error)) for error in errors]
+    }
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+    return [
+        format_single_linter_file(
+            report, linter_report[report]
+        ) for report in linter_report
```

Comment:
```suggestion
        format_single_linter_file(
            file_path, report
        ) for file_path, report in linter_report.items()
```

You can also do something like that, but it's not required :)